src/unix/system.c

```diff
@@ -341,7 +345,27 @@ void Sys_Init(void)
     if (!homedir) {
 	    Sys_Error("Homedir not found!\n");
     }
-    sprintf(homegamedir, "%s/%s", homedir, ".quake2rtx");
+    xdg_data_home_dir = getenv("XDG_DATA_HOME");
+    if (!xdg_data_home_dir) {
```

Comment:
```suggestion
    if (!xdg_data_home_dir) {
        size_t len = strlen(homedir) + strlen("/.local/share") + 1;
```

You use this more than twice, and any desync can cause a memory leak.

Reply:
Done: 7ae9ce53.